src/Xamarin.Android.Build.Tasks/Utilities/LlvmIrGenerator/LlvmIrGenerator.cs

```diff
@@ -981,8 +999,19 @@ void WriteArrayValue (GeneratorWriteContext context, LlvmIrVariable variable)
 					list.Add (kvp.Value);
 				}
 				entries = list;
-			} else {
+			} else if (variable.Type.ImplementsInterface (typeof (ICollection))) {
 				entries = (ICollection)variable.Value;
+			} else if (variable.Type.ImplementsInterface (typeof (ICollection<>))) {
+				// This is slow and messy, but should work for a wide range of types without us having to add
+				// any explicit support
+				Type elementType = variable.Type.GetArrayElementType ();
```

Comment:
I'm just wondering if there is a way we can use System.Metadata to support this?

Reply:
It's possible, but definitely something for a separate PR :) I'm a bit reluctant to use System.Reflection.Metadata, if I'm honest, but not against giving it another try